hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadata.java

```diff
@@ -132,6 +132,11 @@ static HoodieBackedTableMetadata createHoodieBackedTableMetadata(HoodieEngineCon
    */
   FileStatus[] getAllFilesInPartition(Path partitionPath) throws IOException;
 
+  /**
+   * Fetch list of all partitions path that match the given prefix
```

Comment:
Let's make it very clear that the prefix here is referring to "partition-path being the prefix of the complete relative partition path"

Reply:
Improved, also added an example to illustrate the idea.